### PR TITLE
User service

### PR DESCRIPTION
**Pull request by jaredrossberg**
Implements the UserAccountService class and makes necessary changes so that register, login, and logout functionality works as expected.

Makes changes to the API layer to return return expected results and catch various exceptions.

Makes changes to the DAO layer to change session_key to auth_token and return null when nothing is retrieved from the db.

### CODE REVIEW

**Comment by jaredrossberg** [comment]
Choose the squash and merge option when this gets merged to make things a little cleaner

File: src/main/java/com/example/model/Auth.java [hunk 1]
```diff
@@ -10,7 +11,40 @@
 @Data
 @AllArgsConstructor
 public class Auth {
-  private int userId;
-  private String authToken;
-  private Date createTime;
+  public int userId;
```
Comment (cannonj3) [inline src/main/java/com/example/model/Auth.java:16]: The "@Data" annotation automagically creates getters and setters for the data members. The "@AllArgsConstructor" also automatically creates a constructor with all the args so you don't need that either. The other constructors you can add but I would take the rest out.

File: src/main/java/com/example/model/User.java [hunk 2]
```diff
@@ -4,12 +4,72 @@
...
-@AllArgsConstructor
+//@AllArgsConstructor
 public class User {
-  private int userId;
-  private String email;
-  private String password;
-  private String salt;
-  private String displayName;
-  private String phone;
+  public int userId;
```
Comment (cannonj3) [inline src/main/java/com/example/model/User.java:13]: Same comment here as above.

File: src/main/java/com/example/model/Vehicle.java [hunk 3]
```diff
@@ -6,20 +6,149 @@
...
+    this.registrationMonth=registrationMonth;
+    this.mileage=mileage;
+    this.make=make;
+    this.model=model;
+    this.year=year;
+    this.color=color;
+    this.transmissionType=transmissionType;
+    this.energyType=energyType;
+  }
+
```
Comment (cannonj3) [inline src/main/java/com/example/model/Vehicle.java:43]: Same here

File: src/main/java/com/example/model/Auth.java (hunk 1 above)
Comment (jaredrossberg) [reply src/main/java/com/example/model/Auth.java:16]: Good catch. Fixed

**Review by cannonj3 — APPROVED** [review]

**Pull request merged**